Capstone Two Data Wrangling
E Commerce Delivery Performance Prediction

Goal
Predict whether an order will be delivered on time

Unit of analysis
Each row is one shipped order

Target variable
Reached.on.Time_Y.N

In [20]:
import pandas as pd

df = pd.read_csv("Train.csv")

df.shape
df.head()

ID Warehouse_block Mode_of_Shipment  Customer_care_calls  Customer_rating  \
0   1               D           Flight                    4                2   
1   2               F           Flight                    4                5   
2   3               A           Flight                    2                2   
3   4               B           Flight                    3                3   
4   5               C           Flight                    2                2   

   Cost_of_the_Product  Prior_purchases Product_importance Gender  \
0                  177                3                low      F   
1                  216                2                low      M   
2                  183                4                low      M   
3                  176                4             medium      M   
4                  184                3             medium      F   

   Discount_offered  Weight_in_gms  Reached.on.Time_Y.N  
0                44           1233                    1  
1                59           3088                    1  
2                48           3374                    1  
3                10           1177                    1  
4                46           2484                    1

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10999 entries, 0 to 10998
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   ID                   10999 non-null  int64 
 1   Warehouse_block      10999 non-null  object
 2   Mode_of_Shipment     10999 non-null  object
 3   Customer_care_calls  10999 non-null  int64 
 4   Customer_rating      10999 non-null  int64 
 5   Cost_of_the_Product  10999 non-null  int64 
 6   Prior_purchases      10999 non-null  int64 
 7   Product_importance   10999 non-null  object
 8   Gender               10999 non-null  object
 9   Discount_offered     10999 non-null  int64 
 10  Weight_in_gms        10999 non-null  int64 
 11  Reached.on.Time_Y.N  10999 non-null  int64 
dtypes: int64(8), object(4)
memory usage: 1.0+ MB


The dataset contains 10999 shipped orders and 12 features. Warehouse block, shipment mode, product importance, and gender are categorical. Weight, cost, discount, prior purchases, customer rating, and customer care calls are numeric. Reached.on.Time_Y.N is the binary target variable.

In [22]:
df.isnull().sum()

ID                     0
Warehouse_block        0
Mode_of_Shipment       0
Customer_care_calls    0
Customer_rating        0
Cost_of_the_Product    0
Prior_purchases        0
Product_importance     0
Gender                 0
Discount_offered       0
Weight_in_gms          0
Reached.on.Time_Y.N    0
dtype: int64

There are no missing values in any columns. 
Therefore no imputation or row removal was required.

In [ ]:
df.describe(include="all")

In [ ]:
df.duplicated().sum()

No duplicate orders were found so no rows were removed.

In [ ]:
df[df["Weight_in_gms"] <= 0]
df[df["Cost_of_the_Product"] <= 0]

In [ ]:
df[["Weight_in_gms","Cost_of_the_Product","Discount_offered"]].boxplot()

In [ ]:
df["Reached.on.Time_Y.N"].value_counts(normalize=True)

In [24]:
df.to_csv("cleaned_shipping_data.csv", index=False)